patches/minecraft/net/minecraft/block/BlockCactus.java.patch

```diff
@@ -14,24 +14,30 @@
  
          Block block = p_176586_1_.func_180495_p(p_176586_2_.func_177977_b()).func_177230_c();
 -        return block == Blocks.field_150434_aF || block == Blocks.field_150354_m;
-+        return block.canSustainPlant(p_176586_1_, p_176586_2_.func_177977_b(), EnumFacing.UP, this);
++        return block == Blocks.field_150434_aF || net.minecraftforge.common.SoilType.canPlantStayDefault(p_176586_1_, p_176586_2_, this, p_176586_1_.func_180495_p(p_176586_2_));
      }
  
      public void func_180634_a(World p_180634_1_, BlockPos p_180634_2_, IBlockState p_180634_3_, Entity p_180634_4_)
-@@ -137,4 +137,16 @@
+@@ -137,4 +137,22 @@
      {
          return new BlockState(this, new IProperty[] {field_176587_a});
      }
 +
 +    @Override
-+    public net.minecraftforge.common.EnumPlantType getPlantType(net.minecraft.world.IBlockAccess world, BlockPos pos)
++    public java.util.Set<net.minecraftforge.common.SoilType> getSoilTypes(net.minecraft.world.IBlockAccess world, BlockPos pos, IBlockState state)
 +    {
-+        return net.minecraftforge.common.EnumPlantType.Desert;
++        return java.util.Collections.singleton(net.minecraftforge.common.SoilType.Desert);
```

Comment:
This means other desert plants can be planted on top of Cactus. Are you sure this is something we want?